x-pack/plugins/maps/public/classes/styles/vector/components/symbol/icon_map_select.tsx

```diff
@@ -32,6 +37,8 @@ interface StyleOptionChanges {
 interface Props {
   customIconStops?: IconStop[];
   iconPaletteId: string | null;
+  customIcons: CustomIcon[];
+  onCustomIconsChange?: (customIcons: CustomIcon[]) => void;
```

Comment:
Type as `onCustomIconsChange: (customIcons: CustomIcon[]) => void;`

onCustomIconsChange is passed to IconSelect which does not have a guard that onCustomIconsChange is undefined